lib/worker-pipeline.js

```diff
@@ -23,6 +35,13 @@ module.exports = async function ({ targets }) {
         throw error
       }
     }
+
+    // Depending on how the default export is performed, and on how the code is
+    // transpiled, we may find cases of two nested "default" objects.
+    // See https://github.com/pinojs/pino/issues/1243#issuecomment-982774762
+    if (typeof fn === 'object') fn = fn.default
+    if (typeof fn === 'object') fn = fn.default
```

Comment:
The  case where a functions is exported directly is already handled, as `fn` is directly obtained from that import, and we only try this re-assignment in case it's an object.

I'll try to move this into the try block as suggested.

Reply:
I took another look to the code, and I'm not sure it brings any benefit to move these assignments inside the `try` block. If there's anything to be thrown (like `throw Error('exported worker is not a function')`) it would be automatically catch and re-thrown (making the error handling a bit slower), and, besides that, we should duplicate the assignments, because we have a case inside the `catch` block where we use "require" instead of import.